lib/mhv_logging/client.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+require 'faraday'
+require 'common/client/errors'
+require 'common/client/middleware/response/json_parser'
+require 'common/client/middleware/response/raise_error'
+require 'common/client/middleware/response/snakecase'
+require 'mhv_logging/configuration'
+require 'mhv_logging/client_session'
+require 'mhv_logging/api/audits'
+require 'mhv_logging/api/sessions'
+
+module MHVLogging
+  # Core class responsible for api interface operations
+  class Client
+    include MHVLogging::API::Audits
+    include MHVLogging::API::Sessions
+
+    REQUEST_TYPES = %i(get post).freeze
+    USER_AGENT = 'Vets.gov Agent'
+    BASE_REQUEST_HEADERS = {
+      'Accept' => 'application/json',
+      'Content-Type' => 'application/json',
+      'User-Agent' => USER_AGENT
+    }.freeze
+
+    attr_reader :config, :session
+
+    def initialize(session:)
+      @config = MHVLogging::Configuration.instance
+      @session = MHVLogging::ClientSession.find_or_build(session)
```

Comment:
I see what you mean now. Especially since one is a hash and the other is an object federalized from hash having the same name. I will fix this in a larger refactor though if that's OK. 

I will need to do that anyway as part of building a generic client class to out in common::client